app/member-reminders/member-reminders-public/src/main/kotlin/com/hedvig/android/memberreminders/GetConnectPaymentReminderUseCase.kt

```diff
@@ -8,28 +8,37 @@ import com.hedvig.android.apollo.safeExecute
 import com.hedvig.android.apollo.toEither
 import com.hedvig.android.core.common.ErrorMessage
 import com.hedvig.android.logger.logcat
+import kotlinx.datetime.LocalDate
 import octopus.GetPayinMethodStatusQuery
 import octopus.type.MemberPaymentConnectionStatus
 
 internal interface GetConnectPaymentReminderUseCase {
-  suspend fun invoke(): Either<ConnectPaymentReminderError, ShowConnectPaymentReminder>
+  suspend fun invoke(): Either<ConnectPaymentReminderError, PaymentReminder>
 }
 
 internal class GetConnectPaymentReminderUseCaseImpl(
   private val apolloClient: ApolloClient,
 ) : GetConnectPaymentReminderUseCase {
-  override suspend fun invoke(): Either<ConnectPaymentReminderError, ShowConnectPaymentReminder> {
+  override suspend fun invoke(): Either<ConnectPaymentReminderError, PaymentReminder> {
     return either {
-      val payinStatus = apolloClient.query(GetPayinMethodStatusQuery())
+      val result = apolloClient.query(GetPayinMethodStatusQuery())
         .safeExecute()
         .toEither(::ErrorMessage)
         .mapLeft(ConnectPaymentReminderError::NetworkError)
         .bind()
-
-      ensure(payinStatus.currentMember.paymentInformation.status == MemberPaymentConnectionStatus.NEEDS_SETUP) {
-        ConnectPaymentReminderError.AlreadySetup
+      val missingPaymentsContractTerminationDate = result.currentMember.activeContracts
+        .filter { it.terminationDueToMissedPayments }
+        .sortedBy { it.terminationDate }
+        .firstOrNull()?.terminationDate
```

Comment:
nit: just continue with the same chaining style, I think it's easier to parse looking quickly at it
```suggestion
        .firstOrNull()
        ?.terminationDate
```